docs/zkapps/o1js/basic-concepts.md

```diff
@@ -127,25 +126,106 @@ You can think assertions in o1js similar to `require()` function in Solidity.
 
 <!-- prettier-ignore -->
 ```ts
-const x = Field(45);
+let x = Field(45);
+Provable.log(x); // Prints 45
+```
+
+## Other Types and Methods
+
+In addition to the `Field` type, o1js gives you a variety structures to help you build a powerful zkApp. You can learn more about them in the next chapter, [Common Types and Functions](./common-types-and-functions.md).
+
+:::note
+
+The information given until here is more than enough for you to start creating powerful zkApps with o1js. For more advanced level content, you can read the following sections. If you are new to o1js, just continue with the next chapter, [Common Types and Functions](./common-types-and-functions.md).
+
+:::
+
+## Deeper into the Field Type
+
+`Field` element is a positive integer in a prime order [finite field](https://en.wikipedia.org/wiki/Finite_field), which is a set of integers in between 1 and the prime order of the field (a.k.a. an arbitrary prime number describing the field).
+
+The order of the o1js `Field` can be accessed as a static property of the class.
 
-console.log(x.toString()); // string
-console.log(x.toBigInt()); // bigint
-console.log(x.toJSON()); // JSON string
+<!-- prettier-ignore -->
+```ts
+console.log(Field.ORDER);
 ```
 
-However, note that these methods are _not_ provable, meaning you cannot use them in your final zkApp as a part of the ZKP. Use them only to debug your code.
+When you perform an operation inside a finite field, the result always stays in the range. This is achieved through [modular arithmetic](https://en.wikipedia.org/wiki/Modular_arithmetic). This behaviour may be unfamiliar to you if you have not worked with ZKPs before. Let us explain more with some examples.
+
+In the addition or multiplication operations where the result is bigger than the order of the finite field, you receive the modulus of the result with the prime order of the field.
 
 <!-- prettier-ignore -->
 ```ts
-const x = Field(45);
-const y = Field(x.toString()); // You should NOT write this
+Provable.log(Field(Field.ORDER - 1) + Field(2)); // This will print 1
 ```
 
-## Other Types and Methods
+If you try creating a `Field` from a negative number you will receive the modular additive inverse of the number. For instance `(1 + (Field.ORDER - 1)) % Field.ORDER` is 0, so the additive inverse of 1 is `Field.ORDER - 1` in this field.
 
-In addition to the `Field` type, o1js gives you a variety structures to help you build a powerful zkApp. You can learn more about them in the next chapter, [Common Types and Functions](./common-types-and-functions.md).
+<!-- prettier-ignore -->
+```ts
+Provable.log(Field(-1)); // This will print `Field.ORDER - 1`
+```
+
+Similarly, if your substraction results in a negative number you will get the modular additive inverse.
```

Comment:
it's confusing to me that you describe negative numbers / subtraction results in different language than positive overflow. it's the same so should be described in the same language:

"Similarly, if your subtraction results in a negative number, you receive the positive remainder of the result modulo the prime order of the field."